### PR TITLE
Tosin | Feature/delete existing record

### PR DESCRIPTION
**Pull request by fowobi**
I added delete button to summary landing page table

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -26,3 +35,115 @@ const validateUser = (req, res) => {
 
 // endpoint to validate user credentials
 app.get("/login", validateUser);
+
+app.get("/PupilWithTotalScoreAndSupportAllocation", async (req, res) => {
```
Comment (carrejoe3) [inline server/server.js:39]: This should follow the same pattern as other routes, where the code is split into a separate function.

**Comment by carrejoe3** [comment]
Did you run prettier?

**Review by carrejoe3 — CHANGES_REQUESTED** [review]
Great progress, just a few comments 👍

File: client/src/components/LandingPage.jsx [hunk 2]
```diff
@@ -1,78 +1,100 @@
...
-            </td>
-          </tr>
-        </tbody>
-      </table>
+      <h1>Welcome to landing page</h1>
+      <Link to="/form">Add New Pupil</Link>
+      {pupils.length === 0 ? (
+        <p>No pupils data available.</p>
+      ) : (
+        <table className="pupil-table">
```
Comment (alinosratipour) [inline client/src/components/LandingPage.jsx:56]: we can break this component into smaller parts so it is easier to understand and follow

 you can create component like TableHeaderRow.js
 
```
 
 function TableHeaderRow() {
  return (
    <tr>
      <th>Pupil ID</th>
      <th>Pupil Nickname</th>
      <th>Last Update</th>
      <th>Total Score</th>
      <th>Override Score</th>
      <th>Final Support Allocation</th>
      <th>Summary</th>
      <th>Edit</th>
      <th>Delete</th>
    </tr>
  );
}

export default TableHeaderRow;
```

File: client/src/components/LandingPage.jsx [hunk 3]
```diff
@@ -1,78 +1,100 @@
...
+              <th>Total Score</th>
+              <th>Override Score</th>
+              <th>Final Support Allocation</th>
+              <th>Summary</th>
+              <th>Edit</th>
+              <th>Delete</th>
+            </tr>
+          </thead>
+          <tbody>
+            {pupils.map((pupil) => (
```
Comment (alinosratipour) [inline client/src/components/LandingPage.jsx:71]: here you can create component PupilRow.js

```
function PupilRow({ pupil, onDelete }) {
  return (
    <tr key={pupil.pupil_id}>
      <td>{pupil.pupil_id}</td>
      <td>{pupil.pupil_nickname}</td>
      <td>{pupil.last_update}</td>
      <td>{pupil.total_score}</td>
      <td>{pupil.override_score}</td>
      <td>{pupil.final_support_allocation}</td>
      <td>
        <button>Summary</button>
      </td>
      <td>
        <button>Edit</button>
      </td>
      <td>
        <button onClick={() => onDelete(pupil.pupil_id)}>Delete</button>
      </td>
    </tr>
  );
}

export default PupilRow;
```

File: client/src/components/LandingPage.jsx [hunk 4]
```diff
@@ -1,78 +1,100 @@
...
+          prevPupils.filter((pupil) => pupil.pupil_id !== pupilId)
+        );
+      })
+      .catch((error) => {
+        console.error("Error deleting pupil:", error);
+      });
+  };
+  }
+
   return (
```
Comment (alinosratipour) [inline client/src/components/LandingPage.jsx:49]: finally in LandingPage.jsx you can import those components and use them

```
function LandingPage({ pupils, deletePupil }) {
  return (
    <div>
      <h1>Welcome to the landing page</h1>
      <Link to="/form">Add New Pupil</Link>
      {pupils.length === 0 ? (
        <p>No pupils data available.</p>
      ) : (
        <table className="pupil-table">
          <thead>
            <TableHeaderRow />
          </thead>
          <tbody>
            {pupils.map((pupil) => (
              <PupilRow key={pupil.pupil_id} pupil={pupil} onDelete={deletePupil} />
            ))}
          </tbody>
        </table>
      )}
    </div>
  );
}
```

File: client/src/components/LandingPage.jsx
[the remark this answers is not in the document]
Comment (fowobi) [reply client/src/components/LandingPage.jsx:1]: I have correct it, thank for the feed back

[the remark this answers is not in the document]
Comment (fowobi) [reply client/src/components/LandingPage.jsx:1]: correction done

File: server/server.js (hunk 1 above)
Comment (fowobi) [reply server/server.js:39]: Correction Done

File: server/server.js [hunk 5]
```diff
@@ -26,3 +35,115 @@ const validateUser = (req, res) => {
...
+  try {
```
Comment (fowobi) [inline server/server.js:40]: Correction done

**Pull request merged**